src/Core/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -365,5 +410,25 @@ public static void PopulateDbTypeForParameter(KeyValuePair<string, DbConnectionP
                 }
             }
         }
+
+        private void AddStatementIDToMiddlewareContext(string statementId)
+        {
+            HttpContext? httpContext = HttpContextAccessor?.HttpContext;
+            if (httpContext != null)
+            {
+                // locking is because we could have multiple queries in a single http request and each query will be processed in parallel leading to concurrent access of the httpContext.Items.
+                lock (_httpContextLock)
+                {
+                    if (httpContext.Items.TryGetValue(QUERYIDHEADER, out object? currentValue) && currentValue is not null)
+                    {
+                        httpContext.Items[QUERYIDHEADER] = (string)currentValue + ";" + statementId;
```

Comment:
what if the currentValue object cannot be casted to a string?

Reply:
Added a catch block! If there is an error in casting, we won't add the value to httpcontext.